### PR TITLE
Add linter for BIGINT migrations

### PR DESCRIPTION
**Pull request by dmmatson**
Add Rubocop linter to check `add_column`, `change_column`, and `create_table` migrations for violations of "primary key/foreign key must be BIGINT" rule

### CODE REVIEW

File: rubocop.yml [hunk 1]
```diff
@@ -84,3 +84,8 @@ Mavenlint/RequireSlowHelpers:
   Include:
     - spec/unit/**/*spec.rb
     - spec_v2/unit/**/*spec.rb
+
+# Only check for foreign key types in migrations
```
Comment (guanw88) [inline rubocop.yml:88]:
```suggestion
# Only check for primary and foreign key types in migrations
```

File: lib/rubocop/cop/mavenlint/bigint_for_migration_keys.rb [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            add_offense(node, message: PK_VIOLATION_MSG)
+          end
+        end
+
+        def is_integer_column(node)
+          node.children[4].children.first == :integer
+        end
+
+        def is_foreign_id_column(node)
+          node.children[3].children.first.to_s.match(/.*_id/)
```
Comment (guanw88) [inline lib/rubocop/cop/mavenlint/bigint_for_migration_keys.rb:33]: Sorry if this is a basic linter question, but what does it mean to be a "child" of a node here? Is the node the method and the children the params?

**Review by guanw88 — APPROVED** [review]
Have a general question about how linters work, but LGTM! Thanks for putting this together!

File: lib/rubocop/cop/mavenlint/bigint_for_migration_keys.rb (hunk 2 above)
Comment (dmmatson) [reply lib/rubocop/cop/mavenlint/bigint_for_migration_keys.rb:33]: Rubocop linters are presented an AST (abstract syntax tree) that you can parse. For a migration the `up` node in the tree looks like this:

```
(send nil :change_column
  (sym :preferences)
  (sym :subject_id)
  (sym :bigint)
  (hash
    (pair
      (sym :unsigned)
      (false)
    )
    (pair
      (sym :null)
      (true)
    )
  )
)
```

The root `node` in this case is `send nil :change_column`. So `node.children[3]` is `(sym :bigint)`. You have to do `children.children` to get the actual value of the child node, not sure why the `ruby-parse` library works that way.

Comment (dmmatson) [reply lib/rubocop/cop/mavenlint/bigint_for_migration_keys.rb:33]: Used this blog as an inspiration: https://downey.io/blog/writing-rubocop-linters-for-database-migrations/

**Comment by guanw88** [comment]
Will we need to add `ignore-lint` comments on old migrations when this lint rule is merged in?

**Comment by dmmatson** [comment]
> Will we need to add `ignore-lint` comments on old migrations when this lint rule is merged in?

Yep, I will handle that.

**Pull request merged**